source/MaterialXFormat/Util.cpp

```diff
@@ -0,0 +1,116 @@
+//
+// TM & (c) 2017 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXFormat/Util.h>
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+
+namespace MaterialX
+{
+
+string removeExtension(const string& filename)
+{
+  size_t lastDot = filename.find_last_of('.');
+  if (lastDot == string::npos) return filename;
+  return filename.substr(0, lastDot);
+}
+
+bool readFile(const string& filename, string& contents)
```

Comment:
Small note: Looks like your using different format convention for tabs. We use 4 space tabs.

Reply:
Fixed. 😉